status-im/status-react
# Fix keyboard hiding when authorizing send transaction

fixes #21250

### Summary
When confirming the send amount in `...send.input-amount.view` and attempting to authorize with the password in `...send.transaction-confirmation.view`, the `:wallet/get-suggested-routes-successs` event would trigger a re-render which, in turn, hides the keyboard. Since we don't really need to have the routes calculated **after** the user confirmed the amount, we should stop the calculation before reaching the transaction confirmation screen. In case the user goes back to the amount input screen, the route calculation is started again.

Also, added the stopping of suggested routes calculations when closing the send flow (was already there for bridging).

### Testing notes
This change affects both send and bridge.

#### Areas that maybe impacted


##### Functional

- send
- bridge

### Steps to test


- Open Status
- Go to wallet account
- Press send/bridge
- Choose address & token
- Input amount and wait for successful route (currently can only get this on testnet)
- Confirm and proceed to confirmation and authorization with password
- Slide to confirm and input password
- The keyboard shouldn't hide on its own

status: ready
